### ISSUE TITLE
loading parameters and hyper parameters

### ISSUE THREAD

**Issue by randomgambit** [dmlc/xgboost#11719]
Hello

I know this has been asked in the past, but I am unable to find a working solution here. I think I am missing something obvious. A typical use case for XGBranker models is to train them with optuna.

So I specify a couple of parameters that I optimize, then train the final model as

`final_model = xgb.XGBRanker(**myparams, enable_categorical = True)`

Now doing final_model.get_params() correctly returns the params

The problem is that now I save the model as

`final_model.save_model(mymodel.json)`

and when loading it again all params have disappeared 

```
booster = xgb.XGBRanker()
booster.load_model(mymodel.json)
booster.get_params() 
```

retuns None for most of the params

Am I doing something wrong here? Thank you!! 

Using xgb 3.0.5 and sklearn 1.7.1

**Comment by randomgambit** [comment]
of course my objective is to be able to save the model, and the load it to predict a new instance with **exactly** the same set of hyperparameters (including the correct metric and correct nb of trees). Any suggestions on the model xgb-compliant to do so would be greatly appreciated!

**Comment by trivialfis** [comment]
In short: use pickle if you want to serialize the whole thing.

However, the pickle format is *NOT* stable. Any change in the environment, be it the sklearn version, XGBoost version, or Python version, can lead to issues. If you want the hyperparameters, simply use the `get_params` method and save the resulting dictionary.

**Comment by trivialfis** [comment]
Feel free to reopen if there are further questions.
